Src/FluentAssertions/Primitives/StringEqualityValidator.cs

```diff
@@ -27,14 +27,56 @@ protected override bool ValidateAgainstSuperfluousWhitespace()
 
         protected override bool ValidateAgainstLengthDifferences()
         {
+            // Logic is a little bit convoluted because I want to avoid calculation
+            // of mismatch segment in case of equalLength == true for performance reason.
+            // If lazy version of FailWith would be introduced, calculation of mismatch
+            // segment can be moved directly to FailWith's argument
+            bool equalLength = subject.Length == expected.Length;
+
+            string mismatchSegment = GetMismatchSegmentForStringsOfDifferentLengths(equalLength);
+
             return assertion
-                .ForCondition(subject.Length == expected.Length)
+                .ForCondition(equalLength)
                 .FailWith(
-                    ExpectationDescription + "{0} with a length of {1}{reason}, but {2} has a length of {3}.",
+                    ExpectationDescription + "{0} with a length of {1}{reason}, but {2} has a length of {3}, differs near " + mismatchSegment +  ".",
                     expected, expected.Length, subject, subject.Length)
                 .SourceSucceeded;
         }
 
+        private string GetMismatchSegmentForStringsOfDifferentLengths(bool equalLength)
+        {
+            if (equalLength)
```

Comment:
🔧 A special reason why `equalLength` is a parameter, when both `subject` and `expected` are available in the method scope?